src/main/java/org/wso2/lsp4intellij/contributors/annotator/LSPAnnotator.java

```diff
@@ -81,23 +78,51 @@ public void apply(@NotNull PsiFile file, Object annotationResult, @NotNull Annot
         if (FileUtils.isFileSupported(virtualFile) && IntellijLanguageClient.isExtensionSupported(virtualFile)) {
             String uri = FileUtils.VFSToURI(virtualFile);
             EditorEventManager eventManager = EditorEventManagerBase.forUri(uri);
-
-            if (eventManager == null || eventManager.isDiagnosticsLocked()) {
+            if (eventManager == null) {
                 return;
             }
-            try {
-                createAnnotations(file, holder, eventManager);
-            } catch (ConcurrentModificationException e) {
-                // Todo
+
+            if (eventManager.isCodeActionSyncRequired()) {
+                try {
+                    updateAnnotations(holder, eventManager);
+                } catch (ConcurrentModificationException e) {
+                    // Todo
+                } catch (Throwable t) {
+                    LOG.warn("Error occurred when updating LSP diagnostics.", t);
+                }
+            } else if (eventManager.isDiagnosticSyncRequired()) {
+                try {
+                    createAnnotations(holder, eventManager);
+                } catch (ConcurrentModificationException e) {
+                    // Todo
```

Comment:
Let's add a TODO into line 99, to solve concurrent modification issue. Having it here, fails the `createAnnotations()` invocation silently whenever a `ConcurrentModificationException` occurs.

Reply:
+1 and  modified accordingly :)